plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/authorization/AuthorizationMessageConverterV1.java

```diff
@@ -84,13 +85,27 @@ public String verifyConnectionRequestBody(Map<String, String> configuration) {
     }
 
     @Override
-    public String authenticateUserRequestBody(String username, String password) {
+    public String authenticateUserRequestBody(String username, String password, List<SecurityAuthConfig> authConfigs) {
```

Comment:
We are dealing with a Map and string keys in this class. Could we may be use a DTO for representing the data instead? Something like [this](https://github.com/gocd/gocd/blob/5e0a5079f8cc06b1292c006ca40950ca8aea2177/plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/notification/v1/StageNotificationDTO.java) which is used by Notification extension.  Things like `securityAuthConfig.getConfigurationAsMap` are also prone to break as the domain implementation changes.